src/shared/components/data-table/components/main/components/table-row/TableRow.test.tsx

```diff
@@ -0,0 +1,137 @@
+
+const { columns, data } = createDataTableSampleData(1, 10);
+
+const defaultProps = {
+  ...defaultDataTableState,
+  data,
+  dispatch: jest.fn(),
+  columns,
+  rows: data.map((row, index) => ({
+    rowId: index,
+    cells: row
+  }))
+};
+
+describe('<TableRow />', () => {
+  const renderTableRow = (props: Partial<TableContextType> = {}) =>
+    render(
+      <TableContext.Provider value={{ ...defaultProps, ...props }}>
+        <table>
+          <tbody>
+            <TableRow
+              rowId={String(defaultProps.rows[0].rowId)}
+              rowData={defaultProps.rows[0].cells}
+            />
+          </tbody>
+        </table>
+      </TableContext.Provider>
+    );
+
+  let container: HTMLElement;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders without error', () => {
+    container = renderTableRow().container;
+    expect(() => container).not.toThrow();
+  });
+
+  it('respects hiddenRowIds and renders nothing', () => {
+    container = renderTableRow({
+      hiddenRowIds: { [defaultProps.rows[0].rowId]: true }
+    }).container;
+
+    const tableRowElement = container.querySelector('tr');
+    expect(tableRowElement).toBeFalsy();
+  });
+
+  it('respects hiddenColumnIds', () => {
+    container = renderTableRow().container;
+
+    const [firstCellContent, secondCellContent] = defaultProps.rows[0].cells;
+    let firstCellElement = container.querySelector('td');
+    expect(firstCellElement?.textContent).toBe(firstCellContent);
+
+    container = renderTableRow({
+      hiddenColumnIds: { [defaultProps.columns[0].columnId]: true }
+    }).container;
+
+    firstCellElement = container.querySelector('td');
+    expect(firstCellElement?.textContent).toBe(secondCellContent);
+  });
+
+  it('respects individual column renderer prop', () => {
+    const columns = [...defaultProps.columns];
+    const renderer = jest.fn();
+
+    const indexOfColumnWithRenderer = 0;
+    columns[indexOfColumnWithRenderer].renderer = renderer;
+
+    container = renderTableRow({
+      columns
+    }).container;
+
+    expect(renderer).toBeCalledWith({
+      rowData: defaultProps.rows[0].cells,
+      rowId: String(defaultProps.rows[0].rowId),
+      cellData: defaultProps.rows[0].cells[indexOfColumnWithRenderer]
+    });
+  });
+
+  it('respects expandedContent', () => {
+    const firstRowId = defaultProps.rows[0].rowId;
+
+    container = renderTableRow({
+      expandedContent: {
+        [firstRowId]: 'foo'
+      }
+    }).container;
+
+    const expandedContent = container.querySelector('.rowFooter')?.textContent;
+
+    expect(expandedContent).toBe('foo');
+  });
+
+  it('respects searchText', () => {
+    container = renderTableRow({
+      searchText: 'foo'
+    }).container;
+
+    expect(container.querySelector('tr')).toBeFalsy();
+
+    container = renderTableRow({
+      searchText: '1'
+    }).container;
+
+    expect(container.querySelector('tr')).toBeTruthy();
+  });
```

Comment:
I'm uneasy about this. Why is a row responsible for search — why isn't it the table's responsibility?

Reply:
Moved the search handler to `useDataTable` hook.